### PR TITLE
Spacy and Huggingface integrations

### PR DESCRIPTION
**Pull request by adamkells**
## Description
This PR adds convenient integrations for integrating pipelines from spacy, huggingface or langchain into healthchain pipelines. 

These NLP pipelines constitue a single component of the Healthchain pipeline.

## Related Issue
This addresses issue #78 .

## Changes Made

- Modifies the Document container object to be less specific to spacy and to allow addition of information from spacy/hf/langchain.
- Adds integration to instantiate components from each of the three libraries.
- Adds documentation with examples for each integration as well as full example using all 3 in one pipeline.

## Testing
Unit tests were added for all new functionality. As transformers is an optional dependency, the huggingface tests are skipped in CI but were run locally.

## Checklist

- [x] I have read the [contributing](https://github.com/dotimplement/HealthChain/CONTRIBUTING.md) guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by jenniferjiangkells — COMMENTED** [review]
Looks good overall, just left some comments on some stylistic decisions and documentation improvements. Also note updated usage of `add_node()` and importing from `.base`. You good to make changes or want me to take over?

File: docs/reference/pipeline/integrations.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+prompt = PromptTemplate(template=template, input_variables=["text"])
+llm_chain = LLMChain(prompt=prompt, llm=fake_llm)
+langchain_component = LangChainComponent(chain=llm_chain)
+
+# Build our pipeline
+pipeline = Pipeline()
+pipeline.add(spacy_component)
+pipeline.add(huggingface_component)
+pipeline.add(langchain_component)
+
```
[the remark this answers is not in the document]
Comment (adamkells) [reply docs/reference/pipeline/integrations.md:171]: I checked the example and it runs the same with or without the build step, what does this do?

File: healthchain/io/containers.py [hunk 2]
```diff
@@ -86,15 +86,62 @@ class Document(DataContainer[str]):
...
+        ]
+
+    def add_huggingface_output(self, task: str, output: Any):
+        self.huggingface_outputs[task] = output
+
+    def add_langchain_output(self, task: str, output: Any):
+        self.langchain_outputs[task] = output
+
+    def get_tokens(self) -> List[str]:
+        return self.tokens
```
[the remark this answers is not in the document]
Comment (adamkells) [reply healthchain/io/containers.py:145]: How should we decide which attributes to make internal?

File: docs/reference/pipeline/integrations.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+from healthchain.pipeline.components.integrations import HuggingFaceComponent
+
+huggingface_component = HuggingFaceComponent(task="sentiment-analysis", model="distilbert-base-uncased-finetuned-sst-2-english")
+```
+
+
+- `task` (str): The NLP task to perform (e.g., "sentiment-analysis", "named-entity-recognition").
+- `model` (str): The name or path of the Hugging Face model to use.
+
+This component applies the specified Hugging Face model to the input document and stores the output in the HealthChain Document's `huggingface_outputs` dictionary.
```
[the remark this answers is not in the document]
Comment (adamkells) [reply docs/reference/pipeline/integrations.md:73]: Should we just link to the hf models page or should we make clearer which types of models to use?

[the remark this answers is not in the document]
Comment (jenniferjiangkells) [reply docs/reference/pipeline/integrations.md:94]: probably both

File: healthchain/io/containers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jenniferjiangkells) [reply healthchain/io/containers.py:145]: anything that the user shouldn't access directly

File: docs/reference/pipeline/integrations.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+pipeline = Pipeline()
+pipeline.add_node(SpacyComponent(path_to_pipeline="en_core_web_sm"))
+
+doc = Document("This is a test sentence.")
+processed_doc = pipeline(doc)
+
+# Access spaCy annotations
+spacy_doc = processed_doc.get_spacy_doc()
+for token in spacy_doc:https://github.com/dotimplement/HealthChain
```
Comment (jenniferjiangkells) [inline docs/reference/pipeline/integrations.md:76]: ?

**Pull request merged**